design-proposals/isolate-emulator-thread-with-smt-fix/isolate-emulator-thread-with-smt-fix.md

```diff
@@ -0,0 +1,160 @@
+# Preface
+High performance VMs, such as VMs using [DPDK](https://en.wikipedia.org/wiki/Data_Plane_Development_Kit) or [realtime](https://en.wikipedia.org/wiki/Real-time_computing) applications are sensitive to interference from neighboring workloads.
+In order to avoid noisy CPU neighbors, such VMs need to use the [dedicatedCPUPlacement and IsolateEmulatorThread](https://kubevirt.io/user-guide/virtual_machines/dedicated_cpu_resources/#requesting-dedicated-cpu-for-qemu-emulator) features, along with other custom cluster configuration.
+One of these cluster changes is to set the Kubelet’s CPUManager Policy to static - [full-pcpus-only](https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/). This forces kubelet to only accept Pods that have CPUs with full physical cores.
+
+However, on nodes with [hyperthreading](https://en.wikipedia.org/wiki/Hyper-threading) (SMT) enabled, this causes an error, described on [BZ#2228103](https://bugzilla.redhat.com/show_bug.cgi?id=2228103).
+This design proposal is meant to fix this issue.
+
+Considering that hyperthreading (SMT) is enabled by default in many [Intel](https://www.intel.com/content/www/us/en/architecture-and-technology/hyper-threading/hyper-threading-technology.html) processors - it means that DPDK and Realtime users may be impacted.
+
+## Motivation
+Running High performance VMs with an even number of CPUs with the conditions below.
```

Comment:
That includes guests with SMT enabled (spec.domain.cpu.threads: 2).

Reply:
yes.

Reply:
> That's not needed. You can turn off SMT on the guest `spec.domain.cpu.threads: 1`, and still get the SMT issue on the pod.

I gave an example of a specific scenario that is not working.
Any even combination of sockets, cores and threads will cause this problem.

Reply:
I don't follow. Is there something we need to add here to make things clearer?

Reply:
After talking offline - resolving issue